### PR TITLE
RPRBLND-1664: Add support for ML denoiser + upscale

### PR DESCRIPTION
**Pull request by bnagirniak**
### PURPOSE
RIF has upscale ML filter, which provides us to render at lower resolution in viewport and upscaling rendered image.

### EFFECT OF CHANGE
Enabled upscaling in viewport render, which provides faster render at 2-times lower resolution and upscaling rendered image. Added correspondent option in viewport sampling tab.

### TECHNICAL STEPS
1. Added class ImageFilterUpscale.
2. Added Engine.upscale_filter and correspondent methods.
3. Added viewport_resolution_upscale property in user settings, added this property in UI in viewport sampling tab.
4. Implemented upscale_filter usage in ViewportEngine and ViewportEngine2.

### CODE REVIEW

**Comment by bsavery** [comment]
Looks ok I want to review performance and UI next week.

File: src/rprblender/ui/render.py [hunk 1]
```diff
@@ -170,6 +170,8 @@ def draw(self, context):
 
         col.prop(settings, 'use_gl_interop')
 
+        col.prop(settings, 'viewport_resolution_upscale')
```
Comment (ardraeiss) [inline src/rprblender/ui/render.py:173]: Maybe it should be hidden or disabled for Legacy and Hybrid modes?

**Review by ardraeiss — APPROVED** [review]
It's OK, maybe some UI adjustment could be done since this is Full mode only feature.

File: src/rprblender/ui/render.py (hunk 1 above)
Comment (bnagirniak) [reply src/rprblender/ui/render.py:173]: It is working in all modes.

**Review by bsavery — CHANGES_REQUESTED** [review]
I'm going to propose a simple UI change.  Make a button in viewport settings

"Viewport Denoising and upsampling" Which enables both in the viewport.  enabled by default

**Comment by bnagirniak** [comment]
Made one option for viewport denosing and upscaling with ML denosier.

**Review by ardraeiss — APPROVED** [review]

**Pull request merged**